src/v/cluster/topic_configuration.cc

```diff
@@ -65,6 +65,34 @@ storage::ntp_config topic_configuration::make_ntp_config(
       init_rev};
 }
 
+void topic_configuration::serde_write(iobuf& out) {
+    write(out, tp_ns);
+    serde::write(out, partition_count);
```

Comment:
what's `write` vs `serde::write`?

Reply:
https://github.com/redpanda-data/redpanda/blob/ac84d0dd9ecc490e6355689617016ecb0f94e1d1/src/v/model/metadata.h#L368-L372

Commit: 32f521775a49b03e663041233f358c6bf8159dc5

`topic_namespace` is not a serde envelope for some reason, so we hit this friend overload.

Reply:
i don't have a good sense of how the codebase evolved into serde support. @felixguendling might have some historical context?

Reply:
thanks. ill have to look into in more detail to understand. i think i saw a few other places like this too (not in this pr necessarily)

Reply:
There are some non-templated `read_nested` overloads that work this way. `index_state` or something iirc.

Reply:
The reasoning was that there are some data structures that don't need versioning at all because they will never change (such as strings, vectors, etc.). My guess is that we decided that `topic` and a few others won't change and therefore don't need versioning ( = do not derive from `envelope`).

I would always recommend `using serde::write;` - then it doesn't matter if the `write` function is declared in the `serde` namespace or the struct's namespace (and it can even be moved around without changing the serialization code).